sale_payment_method/sale.py

```diff
@@ -0,0 +1,246 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Guewen Baconnier, Sébastien Beau
+#    Copyright (C) 2011 Akretion Sébastien BEAU <sebastien.beau@akretion.com>
+#    Copyright 2013 Camptocamp SA (Guewen Baconnier)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import api, models, fields, exceptions, _
+import openerp.addons.decimal_precision as dp
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.one
+    @api.depends('amount_total', 'payment_ids.credit', 'payment_ids.debit')
+    def _compute_amount(self):
+        paid_amount = sum(line.credit - line.debit
+                          for line in self.payment_ids)
+        self.amount_paid = paid_amount
+        self.residual = self.amount_total - paid_amount
+
+    @api.one
+    @api.depends('payment_ids')
+    def _compute_payment_exists(self):
```

Comment:
I'm using the expression here: 

https://github.com/odoomrp/odoomrp-wip/blob/8.0/mrp_operations_extension/views/mrp_routing_view.xml#L37

and it works for me, but I'm seeing that the field is not of the same type (many2many vs one2many).

Have you tried:
```
attrs="{'invisible': [('payment_ids', '=', [(6, 0, []])]}"
```
or
```
attrs="{'invisible': [('payment_ids[0][2]', '=', [])]}"
```
in case payment_ids is expressed in that form?